### ISSUE TITLE
[Bug]: No save image item shown after upgrading to v4.0.6

### ISSUE THREAD

**Issue by maxhis** [fluttercommunity/plus_plugins#897]
### Platform

iOS

### Plugin

share_plus

### Version

4.0.6

### Flutter SDK

3.0.1

### Steps to reproduce

1. Upgrade `share_plus` to version 4.0.0
2. Share an image with `Share.shareFiles([file_path], mimeTypes: ['image/png'])`
3. No `Save image` item shown.

### Logs

```shell
Probably at least one of the constraints in the following list is one you don't want.
	Try this:
		(1) look at each constraint and try to figure out which you don't expect;
		(2) find the code that added the unwanted constraint or constraints and fix it.
(
    "<NSLayoutConstraint:0x2830b64e0 _UIActivityActionCellTitleLabel:0x131d9eb70.height >= 22   (active)>",
    "<NSLayoutConstraint:0x283089220 UIView:0x131d9d4a0.height >= _UIActivityActionCellTitleLabel:0x131d9eb70.height + 30   (active)>",
    "<NSLayoutConstraint:0x2830e5310 'UIView-Encapsulated-Layout-Height' UIView:0x131d9d4a0.height == 30   (active)>"
)
Will attempt to recover by breaking constraint
<NSLayoutConstraint:0x2830b64e0 _UIActivityActionCellTitleLabel:0x131d9eb70.height >= 22   (active)>
Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.
[ShareSheet] connection invalidated
```
```


### Flutter Doctor

```shell
No issues.
```

**Comment by miquelbeltran** [comment]
Duplicated from https://github.com/fluttercommunity/plus_plugins/issues/870 ?

**Comment by maxhis** [comment]
@miquelbeltran It should not be the same problem. The share menu is shown, missing just the `Save Image` item.

The output logs may be the same, IMO it doesn't directly relate to the issue.

**Comment by Tecmesh** [comment]
Same problem, but mine is on android. i have tested with multiple v4 versions from 4.0.3 to 4.0.6 and same problem. Even the result of executing `shareFilesWithResult `save that it is OK. but the image is not shown where i shared it to.

I am taking a screen shot from a widget with a global key, saving the bytes as a png image, retrieve the image and share through the package, on version ^3.1.0 it worked perfectly.

```
final result = await Share.shareFilesWithResult(
      [imgFile.path],
      mimeTypes: ['image/*'],
    );
```

This is my flutter doctor result:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.5, on Microsoft Windows [Version 10.0.19043.1706], locale es-ES)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Chrome - develop for the web
[✓] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.15)
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.67.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

**Comment by maxhis** [comment]
Another issue: on iOS 12.5.5,  tapping`Save Image` item does not save the image to the album, even getting the `ShareResultStatus.success` result.

**Comment by miquelbeltran** [comment]
Thanks a lot for verifying that! I will have to revert the changes in 4.0.6 then.

> Another issue: on iOS 12.5.5, tappingSave Image item does not save the image to the album, even getting the ShareResultStatus.success result.

If it is not a problem, place create another ticket for that issue, so it can be investigated accordingly

**Comment by maxhis** [comment]
> Thanks a lot for verifying that! I will have to revert the changes in 4.0.6 then.
> 
> > Another issue: on iOS 12.5.5, tappingSave Image item does not save the image to the album, even getting the ShareResultStatus.success result.
> 
> If it is not a problem, place create another ticket for that issue, so it can be investigated accordingly

Thanks for response, created a new issue: https://github.com/fluttercommunity/plus_plugins/issues/899

**Comment by Tecmesh** [comment]
I have kinda the same problem, but mine is on android. i have tested with multiple v4 versions from 4.0.3 to 4.0.7 and same problem. Even the result of executing shareFilesWithResult save that it is OK. but the image is not shown where i shared it to.

I am taking a screen shot from a widget with a global key, saving the bytes as a png image, retrieve the image and share through the package, on version ^3.1.0 it worked perfectly.

final result = await Share.shareFilesWithResult(
      [imgFile.path],
      mimeTypes: ['image/*'],
    );
This is my flutter doctor result:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.5, on Microsoft Windows [Version 10.0.19043.1706], locale es-ES)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Chrome - develop for the web
[✓] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.11.15)
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.67.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

**Comment by miquelbeltran** [comment]
@Tecmesh please create a new ticket with your issue

**Comment by Coronon** [comment]
Hi there, I was the author of version `4.0.6`, could you send me a screenshot where the 'save image' option is not displayed? Everything works as expected on my IPhone 12 Pro Max.

**Comment by Coronon** [comment]
Btw. the 'mineTypes' option for images was pretty much ignored in `4.0.6` because it dynamically detected them, I'm guessing that your iOS version now uses some different image types which are not correctly picked up (Im still running iOS 14 ^^).

**Comment by Coronon** [comment]
Sorry to spam here, but I jsut noticed #899, which talks about iOS 12 - `4.0.6` depends on iOS 13 SDK API's. This should be fine as [Apple made them a requiremend for submissions](https://developer.apple.com/news/?id=03042020b) in 2020.

**Comment by maxhis** [comment]
@Coronon Here is the screenshot with `4.0.5 `while sharing an image:

<img width="452" alt="image" src="https://user-images.githubusercontent.com/1464097/173212462-27ed0bee-9b45-4efd-90cd-e691d0985dd8.png">

And here is the same action with `4.0.6`:

<img width="452" alt="image" src="https://user-images.githubusercontent.com/1464097/173212556-1a83bdb0-e411-4002-9d40-f02f4e333da7.png">

**Comment by Coronon** [comment]
Reproducible, I think I found the fix within `activityViewControllerPlaceholderItem`, which was already implemented in the before abandoned `SharePlusData` class.

Will open a PR asap - thanks for raising this issue @maxhis :)

**Comment by Coronon** [comment]
As always, I provide a working example [here](https://github.com/Coronon/share_plus_example).

Please try my example (override your dependencies like I did in my example) and report if this is fixed for you :)

**Comment by maxhis** [comment]
@Coronon Tried in my project with your repo and branch,  works both on the actual device and simulator. Thanks for your efforts.

Just added the following lines in `pubspec.yml`:

```yml
dependency_overrides:
  share_plus:
    git:
      url: https://github.com/Coronon/plus_plugins
      ref: ab99d3c46dcb4ee5c357839c8db2dde66cce18e6
      path: packages/share_plus/share_plus
```

**Comment by Coronon** [comment]
@maxhis Awesome! Must have bricked my simulator somehow... We are ready for merge and just waiting for a review then.

**Comment by miquelbeltran** [comment]
fix released in 4.0.8. As always, thanks a lot @Coronon !!!
